app/services/pricing/service.go

```diff
@@ -58,38 +53,36 @@ func NewService(bridgeConfig config.Bridge,
 	mirrorNodeClient client.MirrorNode,
 	coinGeckoClient client.Pricing,
 	coinMarketCapClient client.Pricing) *Service {
-	initOnce.Do(func() {
-		tokensPriceInfo := make(map[uint64]map[string]pricing.TokenPriceInfo)
-		minAmountsForApi := make(map[uint64]map[string]string)
-		for networkId := range constants.NetworksById {
-			tokensPriceInfo[networkId] = make(map[string]pricing.TokenPriceInfo)
-			minAmountsForApi[networkId] = make(map[string]string)
-		}
+	tokensPriceInfo := make(map[uint64]map[string]pricing.TokenPriceInfo)
+	minAmountsForApi := make(map[uint64]map[string]string)
+	for networkId := range constants.NetworksById {
+		tokensPriceInfo[networkId] = make(map[string]pricing.TokenPriceInfo)
+		minAmountsForApi[networkId] = make(map[string]string)
+	}
 
-		logger := config.GetLoggerFor("Pricing Service")
-		hbarFungibleAssetInfo, _ := assetsService.FungibleAssetInfo(constants.HederaNetworkId, constants.Hbar)
-		hbarNativeAsset := assetsService.FungibleNativeAsset(constants.HederaNetworkId, constants.Hbar)
-		instance = &Service{
-			tokensPriceInfo:       tokensPriceInfo,
-			minAmountsForApi:      minAmountsForApi,
-			mirrorNodeClient:      mirrorNodeClient,
-			coinGeckoClient:       coinGeckoClient,
-			coinMarketCapClient:   coinMarketCapClient,
-			tokenPriceInfoMutex:   new(sync.RWMutex),
-			minAmountsForApiMutex: new(sync.RWMutex),
-			assetsService:         assetsService,
-			coinGeckoIds:          bridgeConfig.CoinGeckoIds,
-			coinMarketCapIds:      bridgeConfig.CoinMarketCapIds,
-			hbarFungibleAssetInfo: hbarFungibleAssetInfo,
-			hbarNativeAsset:       hbarNativeAsset,
-			logger:                logger,
-		}
+	logger := config.GetLoggerFor("Pricing Service")
+	hbarFungibleAssetInfo, _ := assetsService.FungibleAssetInfo(constants.HederaNetworkId, constants.Hbar)
+	hbarNativeAsset := assetsService.FungibleNativeAsset(constants.HederaNetworkId, constants.Hbar)
+	instance := &Service{
+		tokensPriceInfo:       tokensPriceInfo,
+		minAmountsForApi:      minAmountsForApi,
+		mirrorNodeClient:      mirrorNodeClient,
+		coinGeckoClient:       coinGeckoClient,
+		coinMarketCapClient:   coinMarketCapClient,
+		tokenPriceInfoMutex:   new(sync.RWMutex),
+		minAmountsForApiMutex: new(sync.RWMutex),
+		assetsService:         assetsService,
+		coinGeckoIds:          bridgeConfig.CoinGeckoIds,
+		coinMarketCapIds:      bridgeConfig.CoinMarketCapIds,
+		hbarFungibleAssetInfo: hbarFungibleAssetInfo,
+		hbarNativeAsset:       hbarNativeAsset,
+		logger:                logger,
+	}
 
-		err := instance.FetchAndUpdateUsdPrices(true)
-		if err != nil {
-			logger.Fatalf(err.Error())
-		}
-	})
+	err := instance.FetchAndUpdateUsdPrices()
+	if err != nil {
+		panic(err.Error())
```

Comment:
Could we use `log.Fatal` showcasing the error message `Failed to initially fetch USD prices. Error: [%s]`?